File: app/src/main/java/fr/free/nrw/commons/location/LocationUpdateListener.java
package fr.free.nrw.commons.location;

import fr.free.nrw.commons.data.models.location.LatLng;

public interface LocationUpdateListener {
    void onLocationChangedSignificantly(LatLng latLng); // Will be used to update all nearby markers on the map
    void onLocationChangedSlightly(LatLng latLng); // Will be used to track users motion
    void onLocationChangedMedium(LatLng latLng); // Will be used updating nearby card view notification
}

File: app/src/main/java/fr/free/nrw/commons/location/LocationUpdateListener.kt
package fr.free.nrw.commons.location

interface LocationUpdateListener {
    // Will be used to update all nearby markers on the map
    fun onLocationChangedSignificantly(latLng: LatLng?)

    // Will be used to track users motion
    fun onLocationChangedSlightly(latLng: LatLng?)

    // Will be used updating nearby card view notification
    fun onLocationChangedMedium(latLng: LatLng?)
}